src/monaca.js

```diff
@@ -759,6 +759,84 @@
    * @method
    * @memberof Monaca
    * @description
+   *   Checks if the project can be built.
+   * @return {Promise}
+   * @example
+   *   monaca.checkBuildAvailability('myProjectID', 'android', 'debug').then(
+   *     function() {
+   *       //Build the project
+   *     },
+   *     function(err) {
+   *       //Cannot build the project
+   *     }
+   *   );
+   */
+  Monaca.prototype.checkBuildAvailability = function(projectId, platform, buildType) {
+    var deferred = Q.defer();
+
+    if (typeof platform == 'undefined') {
+      deferred.resolve();
+      return deferred.promise;
+    } else if (typeof buildType === 'undefined') {
+      deferred.reject(new Error("Missing --build-type parameter."));
+      return deferred.promise;
+    }
+
+    this._get('/project/' + projectId + '/can_build_app')
+    .then(
+      function(response) {
+        try {
+          var data = JSON.parse(response);
+        } catch (err) {
+          return deferred.reject(new Error(JSON.stringify(err)));
+        }
+
+        if (data.status === 'ok') {
+          var platformContent = data.result[platform];
+
+          if (!platformContent) {
+            return deferred.reject(new Error("Specified platform is not supported."));
+          } else if (platformContent.has_remaining_slot !== true) {
+            return deferred.reject(new Error("Your plan does not allow further builds at the moment. Please upgrade your account to start build, or try again later."));
+          } else if (platformContent.is_start_file_exist !== true) {
+            return deferred.reject(new Error("Your project is missing the start up file (usually index.html)."));
+          } else if (platformContent.manifest_error) {
+            return deferred.reject(new Error("Your AndroidManifest.xml has an invalid value. Please fix it and try again."));
+          } else if (platformContent.can_build_for[buildType] !== true) {
+            return deferred.reject(new Error(platform + " " + buildType + " build is not supported."));
```

Comment:
I think this condition can be deleted: Watanabe san said it's related to other issues, like pslist error, missing icons, provisioning file error etc. As we already check all these conditions, we don't need to check this generalization.